src/main/java/com/textmessenger/service/LoginServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package com.textmessenger.service;
+
+import com.textmessenger.model.entity.User;
+import com.textmessenger.model.entity.dto.LoginRq;
+import com.textmessenger.model.entity.dto.Token;
+import com.textmessenger.repository.UserRepository;
+import com.textmessenger.security.JwtTokenProvider;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+
+public class LoginServiceImpl implements LoginService {
+  private final UserRepository userRepository;
+  private final AuthenticationManager authenticationManager;
+  private final JwtTokenProvider jwtTokenProvider;
+  private final PasswordEncoder passwordEncoder;
+
+  public LoginServiceImpl(PasswordEncoder passwordEncoder,
+                          UserRepository userRepository,
+                          AuthenticationManager authenticationManager,
+                          JwtTokenProvider jwtTokenProvider) {
+    this.userRepository = userRepository;
+    this.authenticationManager = authenticationManager;
+    this.jwtTokenProvider = jwtTokenProvider;
+    this.passwordEncoder = passwordEncoder;
+  }
+
+  @Override
+  public ResponseEntity authenticateUser(LoginRq user) {
+    Authentication authenticate = authenticationManager.authenticate(
+            new UsernamePasswordAuthenticationToken(
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unused "str" local variable. ([squid:S1481](https://sonarcloud.io/organizations/final_project_text_messenger/rules#rule_key=squid%3AS1481))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=Text_Messenger_Project&pullRequest=142&issues=AWWFkfgzX2sY87p0v-Gr&open=AWWFkfgzX2sY87p0v-Gr)

[ISSUE_KEY]: AWWFkfgzX2sY87p0v-Gr